tests/cases/fourslash/noErrorsAfterCompletionsRequestWithinGenericFunction3.ts

```diff
@@ -0,0 +1,25 @@
+///<reference path="fourslash.ts"/>
+// @strict: true
+////
+//// // repro from #52580#issuecomment-1416131055
+////
+//// type Funcs<A, B extends Record<string, unknown>> = {
+////   [K in keyof B]: {
+////     fn: (a: A, b: B) => void;
+////     thing: B[K];
+////   }
+//// }
+////
+//// function foo<A, B extends Record<string, unknown>>(fns: Funcs<A, B>) {}
+////
+//// foo({
+////   bar: { fn: (a: string, b) => {}, thing: "asd" },
+////   /*1*/
+//// });
+
+goTo.marker("1");
+const markerPosition = test.markers()[0].position;
+edit.paste(`bar: { fn: (a: string, b) => {}, thing: "asd" },`)
+edit.replace(markerPosition + 4, 1, 'z')
+verify.completions({ isNewIdentifierLocation: true });
+verify.noErrors();
```

Comment:
This was an interesting case because the whole argument got blocked and thus `A` and `B` were replaced with their constraints and thus typing the argument `Funcs<unknown, Record<string, unknown>>`. And then the argument became not assignable to this target type - see the playground [here](https://www.typescriptlang.org/play?ssl=22&ssc=15&pln=22&pc=54#code/C4TwDgpgBAYgrgOwMYGcA8BBANFAQlCAD2AgQBMUoAlCJAewCcy0VgGBLBAcx0QGsEdAO4IAfKKgBeKAG8AUFCgBtANJROUPhBB0AZngC6ALlkLFUXQhMAKAIYnsUAEYncASikSAbnXZkA3GaKwAAWnFyuqgaBigC+gfFycmS0ADa2DNC6iEjA7HQIFnR0mDj4RCTklDT0TCxs4bwIAsJiotZmligm8Mjojriicm4mSgPRSaCQUADKdHAMSNDS8opOGSar5pY29lCsHNw4LrLOG-sN3P5nAF4mB+FQsR6S3r4BQVCh4feXXDFPAHrO6mcwWKxQOy-Q48ZybM4MaHha7ApHcJ4vN5+AHBMLcNH-MyJRJTaAAFQyXAgwCksBy6H4ghEOBqjGYDyOUEZrXEgTkpKgAGEQrQ+LS5gslgRiKQKFAKQwqTSAPxQACMUBMAAZAgB6XXmAB6yrkQA)